app/components/common/eventModal/index.tsx

```diff
@@ -55,21 +57,24 @@ export default function EventModal({ events, currentEvent, setCalendarEvent }: E
     return resultArray;
   };
 
-  const updateEventStateFromModal = (event: CalEvent) => {
-    setCalendarEvent((e) => ({ ...e, event }));
+  const updateEventStateFromModal = (ev: CalEvent) => {
+    setCalendarEvent((e) => ({ ...e, ev }));
   };
 
-  React.useEffect(() => {
+  useEffect(() => {
     if (params.eventId === 'new' && routerLocation.state) {
       setCalendarEvent((e) => ({
         ...e,
-        start: dayjs(routerLocation.state.start),
-        end: dayjs(routerLocation.state.end),
+        event: {
+          ...e.event,
+          start: dayjs(routerLocation.state.start).toDate(),
```

Comment:
how was this working previously, if we were not converting to date

Reply:
@isVivek99 first of all it was not set inside the `event` property as it should.
After I set it to the event, ts threw the actual error regarding datatype mismatch. That's why the conversion.